src/main/java/org/jenkinsci/plugins/codesonar/services/AlertsService.java

```diff
@@ -0,0 +1,82 @@
+package org.jenkinsci.plugins.codesonar.services;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.http.client.utils.URIBuilder;
+import org.jenkinsci.plugins.codesonar.CodeSonarPluginException;
+import org.jenkinsci.plugins.codesonar.models.CodeSonarAlertData;
+import org.jenkinsci.plugins.codesonar.models.CodeSonarAlertFrequencies;
+
+import com.google.common.base.Throwables;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+public class AlertsService {
+    private static final Logger LOGGER = Logger.getLogger(AlertsService.class.getName());
+    
+    final private HttpService httpService;
+
+    public AlertsService(HttpService httpService) {
+        this.httpService = httpService;
+    }
+    
+    public CodeSonarAlertFrequencies getAlertFrequencies(URI baseHubUri, long analysisId) throws IOException {
+        CodeSonarAlertFrequencies frequencies = new CodeSonarAlertFrequencies();
+        //Loop through all possible kind values, currently from 0 to 26.
+        for(long kind = 0; kind < 26; kind++) {
+            //Build char_table request URL
+            URIBuilder uriBuilder = new URIBuilder(baseHubUri);
+            uriBuilder.setPath(String.format("/analysis/%d-alert%d.json", analysisId, kind));
+            
+            URI requestUri = null;
+            try {
+                requestUri = uriBuilder.build();
+            } catch (URISyntaxException e) {
+                LOGGER.log(Level.WARNING, "Request URI for Alerts API contains a syntax error. %nException: {0}%nStack Trace: {1}", new Object[] {e.getMessage(), Throwables.getStackTraceAsString(e)});
+                return null;
```

Comment:
Why is the caller required to handle it?  Is it because the `throws` clause of a parent method does not allow this type?  Why not convert to a compatible exception type and throw?  When I said "Didn't we make the SyntaxError", I was trying to say that if we get a SyntaxError here, is it not our own fault?

Overall, it seems like returning NULL in so many places in this method could lead to legitimate errors being overlooked.  I like that the log mentions the problems, but it is the system log, and most of the time, no one is going to look there.